### PR TITLE
ethclient, swarm: Correct Subscribe signature

### PR DESCRIPTION
**Pull request by nolash**
Ethclient used old rpc.Client.EthSubscribe signature.

Removed obsolete call to p2p.Server.JoinHostPort in swarm.go (so it can compile)

### CODE REVIEW

File: ethclient/ethclient.go [hunk 1]
```diff
@@ -251,7 +251,7 @@ func (ec *Client) SyncProgress(ctx context.Context) (*ethereum.SyncProgress, err
 // SubscribeNewHead subscribes to notifications about the current blockchain head
 // on the given channel.
 func (ec *Client) SubscribeNewHead(ctx context.Context, ch chan<- *types.Header) (ethereum.Subscription, error) {
-	return ec.c.EthSubscribe(ctx, ch, "newHeads", map[string]struct{}{})
```
Comment (nolash) [inline ethclient/ethclient.go:256]: Strange; wasn't this signature changed a long while ago already?

**Comment by lmars** [comment]
@nolash why is this change necessary? We shouldn't be dropping `EthSubscribe`

File: swarm/swarm.go [hunk 2]
```diff
@@ -218,7 +218,8 @@ func (self *Swarm) Start(net *p2p.Server) error {
 
 	// start swarm http proxy server
 	if self.config.Port != "" {
-		addr := net.JoinHostPort(self.config.ListenAddr, self.config.Port)
+		//addr := net.JoinHostPort(self.config.ListenAddr, self.config.Port)
+		addr := fmt.Sprintf("%s:%d", self.config.ListenAddr, self.config.Port)
```
Comment (lmars) [inline swarm/swarm.go]: Why change this?

**Comment by nolash** [comment]
I didn't change them. The WERE changed in recent commits, I just adapted the calling code so that it doesn't break. I was alerted by this:

https://github.com/ethersphere/go-ethereum/pull/115#issuecomment-319009039

**Comment by lmars** [comment]
Ok I think it was a mistake to drop `EthSubscribe`. [This commit](https://github.com/ethersphere/go-ethereum/commit/a9c8097) from @zelig seems to be the culprit, I think we should revert the branch back and keep the changes separate for now.

**Comment by nolash** [comment]
I concur. It seems a number of things have gone haywire with that commit.

**Pull request closed without merging**